pkg/test/masteruserrecord/master_user_record.go

```diff
@@ -139,12 +134,12 @@ func Namespace(nsType, revision string) UaInMurModifier {
 		for uaIndex, ua := range mur.Spec.UserAccounts {
 			if ua.TargetCluster == targetCluster {
 				for nsIndex, ns := range mur.Spec.UserAccounts[uaIndex].Spec.NSTemplateSet.Namespaces {
-					if ns.Type == nsType {
-						mur.Spec.UserAccounts[uaIndex].Spec.NSTemplateSet.Namespaces[nsIndex].Revision = revision
-						mur.Spec.UserAccounts[uaIndex].Spec.NSTemplateSet.Namespaces[nsIndex].TemplateRef = mur.Spec.UserAccounts[uaIndex].Spec.NSTemplateSet.TierName + "-" + nsType + "-" + revision
+					if strings.Contains(ns.TemplateRef, nsType) {
+						templateRef := strings.ToLower(fmt.Sprintf("%s-%s-%s", mur.Spec.UserAccounts[uaIndex].Spec.NSTemplateSet.TierName, nsType, revision))
```

Comment:
we could also use https://github.com/codeready-toolchain/host-operator/blob/35c9bc260618385fc7b5eb6ea3e9186502afe531/pkg/templates/nstemplatetiers/nstemplatetier_generator.go#L215-L217, but for now it's ok to keep it as-is here, since this func is for testing purpose.

Reply:
@xcoulon good idea! sounds good!